theia-extensions/viewer-prototype/src/node/trace-server-service.ts

```diff
@@ -1,13 +1,23 @@
 import { spawn } from 'child_process';
+import { join } from 'path';
+import { existsSync } from 'fs';
 import { injectable } from 'inversify';
 import { PortBusy, TraceServerConfigService } from '../common/trace-server-config';
+import { FileOperationError, FileOperationResult } from '@theia/filesystem/lib/common/files';
 import treeKill = require('tree-kill');
 @injectable()
 export class TraceServerServiceImpl implements TraceServerConfigService {
     private processId: number;
 
     startTraceServer(path: string | undefined, port: number | undefined): Promise<string> {
-        const server = spawn(`${path}`, ['-vmargs', `-Dtraceserver.port=${port}`]);
+        if (typeof path === 'undefined') {
+            throw new FileOperationError('The Trace Viwer path is not defined.', FileOperationResult.FILE_INVALID_PATH);
+        }
+        const serverPath = join(__dirname, '../../../../', path);
```

Comment:
with this it will be required that the server is relative to the patch that is configure for it. This will only work in what you want to achieve in this patch, but breaks other setups.

The path is configured in a preference so that it can be changed by the user.

Reply:
you're right if the path is absolute, it will not work. However, for all setups this will default to the main directory of where the theia-trace-extension is executed. I can change the implementation so that it takes care of absolute path correctly. Are there any thing else to be taken into account ?

Reply:
We need to be able to handle a couple of contingencies here. Whether or not the server is bundled with the app image, it should be possible for the user to specify a path to their preferred executable.

I suggest the following:
 - Remove the default for the preference specifying the location of the server executable when the application is packaged, but not when it is built locally.
 - Allow `path` to be `undefined` - i.e. don't throw an invalid path error immediately.
 - If `path` is `undefined` or an empty string, use the expected packaged directory.

Reply:
And I'm suggesting that that value be removed from the version of the `package.json` included with the packaged electron application. When the application is packaged, the default will be to use the built-in binary, rather than an external program. Since that behavior is different from the unpackaged version, a change needs to be made during the packaging process.